packages/app/src/cli/services/info.ts

```diff
@@ -149,8 +149,15 @@ class AppInfo {
     const toplevel = ['📂 web', '']
     const sublevels: [string, string][] = []
     this.app.webs.forEach((web) => {
-      if (web.configuration && web.configuration.type) {
-        sublevels.push([`  📂 ${web.configuration.type}`, relativePath(this.app.directory, web.directory)])
+      if (web.configuration) {
+        if (web.configuration.name) {
+          const {name, roles} = web.configuration
+          sublevels.push([`    📂 ${name} (${roles.join(',')})`, relativePath(this.app.directory, web.directory)])
+        } else {
+          web.configuration.roles.forEach((role) => {
+            sublevels.push([`    📂 ${role}`, relativePath(this.app.directory, web.directory)])
+          })
+        }
       } else if (this.app.errors) {
         const error = this.app.errors.getError(`${web.directory}/${configurationFileNames.web}`)
         if (error) {
```

Comment:
could we do a little refactor on this just to avoid nested `if` ?

```suggestion
      if (this.app.errors && !web.configuration) {
        const error = this.app.errors.getError(`${web.directory}/${configurationFileNames.web}`)
        if (error) {
          sublevels.push([`  📂 ${UNKNOWN_TEXT}`, relativePath(this.app.directory, web.directory)])
          errors.push(error)
        }
      }

      if (web.configuration.name) {
        const {name, roles} = web.configuration
        sublevels.push([`    📂 ${name} (${roles.join(',')})`, relativePath(this.app.directory, web.directory)])
      } else {
        web.configuration.roles.forEach((role) => {
          sublevels.push([`    📂 ${role}`, relativePath(this.app.directory, web.directory)])
        })
      }
```

Reply:
This refactor isn't the logical equivalent of what was there before. It really is an either-or. Either we identified the web properly and display information, or there was an issue identifying the web and we display `unknown`.

Reply:
That said, I think my logic is flawed too. If there's `web.configuration` but it has no type/role, the new logic doesn't do the right thing. I'll have to fix.

Reply:
> )

You are right I missed in the second if, sorry :( 

```ts 
if (web.configuration && web.configuration.name)
```

Reply:
Fixed in https://github.com/Shopify/cli/pull/2083/commits/dd4c444a664e18154b490e249fe92f3ce624067b